homeassistant/components/stookwijzer/__init__.py

```diff
@@ -2,29 +2,68 @@
 
 from __future__ import annotations
 
+import logging
+
 from stookwijzer import Stookwijzer
 
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_LATITUDE, CONF_LOCATION, CONF_LONGITUDE, Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
 from .const import DOMAIN
+from .coordinator import StookwijzerCoordinator
 
 PLATFORMS = [Platform.SENSOR]
 
+_LOGGER = logging.getLogger(__name__)
+
+type StookwijzerConfigEntry = ConfigEntry[StookwijzerCoordinator]
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Stookwijzer from a config entry."""
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = Stookwijzer(
-        entry.data[CONF_LOCATION][CONF_LATITUDE],
-        entry.data[CONF_LOCATION][CONF_LONGITUDE],
+    client = Stookwijzer(
+        async_get_clientsession(hass),
+        entry.data[CONF_LATITUDE],
+        entry.data[CONF_LONGITUDE],
     )
```

Comment:
There's no need to create this object here, it can be created in the coordinator

Reply:
Agreed, it isn't in this place in the upcoming refactor.